src/backend/db/database_manager.py

```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
+
+        try:
+            session = DatabaseManager._session_factory()
+            # Add retry logic here for connection
+        except OperationalError as e:
+            logger.error(f"Database connection failed: {e}")
+            return {"status": "db_unreachable", "reason": "Database is currently unreachable"}
+
+        try:
+            # Check if the database exists
+            logger.debug(f"Checking if database {customer_db_name} exists")
+            try:
+                result = session.execute(
+                    text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                    {"db_name": customer_db_name}
+                ).fetchone()
+            except (OperationalError, DatabaseError) as e:
+                logger.error(f"Database connectivity issue during schema check: {e}")
+                return {"status": "db_unreachable", "reason": "Database is currently unreachable"}
+
+            if not result:
+                return {"status": "unknown_db", "reason": f"Database {customer_db_name} does not exist"}
+
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Check if the comment exists
+            result = session.execute(
+                text("SELECT COUNT(*) FROM ticket_comments WHERE ticket_id = :ticket_id AND comment_id = :comment_id"),
+                {"ticket_id": ticket_id, "comment_id": comment_id},
+            ).fetchone()
+
+            if result[0] == 0:
+                return {"status": "not_found",
+                        "reason": f"Comment ID {comment_id} not found for Ticket ID {ticket_id}."}
+
+            # Update the comment and set is_edited to true
+            try:
+                query = """
+                    UPDATE ticket_comments
+                    SET comment = :comment,
+                        is_edited = true
+                    WHERE ticket_id = :ticket_id AND comment_id = :comment_id
+                """
+                session.execute(
+                    text(query),
+                    {"ticket_id": ticket_id, "comment_id": comment_id, "comment": comment_update.comment}
+                )
+
+                # Fetch updated_at column after update
+                updated_at_result = session.execute(
+                    text(
+                        "SELECT updated_at, is_edited FROM ticket_comments WHERE ticket_id = :ticket_id AND comment_id = :comment_id"),
+                    {"ticket_id": ticket_id, "comment_id": comment_id}
+                ).fetchone()
+
+                session.commit()
+
+                updated_at = updated_at_result[0] if updated_at_result else None
+                is_edited = updated_at_result[1] if updated_at_result else False
+
+                return {"status": "updated", "reason": None, "comment": comment_update.comment, "ticket_id": ticket_id, "comment_id":comment_id}
```

Comment:
CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

Reply:
modified sir